src/test/java/org/phoebus/olog/docker/OlogLogsIT.java

```diff
@@ -190,14 +195,52 @@ void handleLogCreateCheck() {
         //         Upload attachment
         //         Upload multiple attachments
 
-    	Log log_check = new Log.LogBuilder().build();
+        Log log_check = new Log.LogBuilder().build();
 
-    	ITUtilLogs.assertListLogs(0);
+        int length = ITUtilLogs.assertListLogs(-1).length;
 
-    	ITUtilLogs.assertCreateLog(AuthorizationChoice.ADMIN, "", log_check, HttpURLConnection.HTTP_BAD_REQUEST);
-    	ITUtilLogs.assertCreateLog(AuthorizationChoice.ADMIN, "", log_check, HttpURLConnection.HTTP_BAD_REQUEST);
+        ITUtilLogs.assertCreateLog(AuthorizationChoice.ADMIN, "", log_check, HttpURLConnection.HTTP_BAD_REQUEST);
+        ITUtilLogs.assertCreateLog(AuthorizationChoice.ADMIN, "", log_check, HttpURLConnection.HTTP_BAD_REQUEST);
 
-    	ITUtilLogs.assertListLogs(0);
+        assertEquals(length, ITUtilLogs.assertListLogs(-1).length);
+    }
+
+    @Test
+    void handleLogGroup() {
+    	// create logbook and prepare logs
+    	Logbook logbook1 = new Logbook("name1", "user", State.Active);
+    	ITUtilLogbooks.assertCreateLogbook("/" + logbook1.getName(), logbook1);
+
+    	Log log1 = LogBuilder.createLog()
+    			.owner("owner")
+    			.title("title1")
+    			.withLogbooks(Set.of(logbook1))
+    			.build();
+    	Log log2 = LogBuilder.createLog()
+    			.owner("user")
+    			.title("title2")
+    			.withLogbooks(Set.of(logbook1))
+    			.build();
+
+    	// create logs
+    	log1 = ITUtilLogs.assertCreateLog("", log1);
+    	log2 = ITUtilLogs.assertCreateLog("", log2);
+    	assertEquals(0, log1.getProperties().size());
+    	assertEquals(0, log2.getProperties().size());
+
+    	// group logs
+    	ITUtilLogs.assertGroupLogs(Arrays.asList(log1.getId(), log2.getId()));
+    	log1 = ITUtilLogs.assertRetrieveLog("/" + log1.getId());
+    	log2 = ITUtilLogs.assertRetrieveLog("/" + log2.getId());
+    	assertEquals(1, log1.getProperties().size());
+    	assertEquals(1, log2.getProperties().size());
+
+    	// edit and update logs but log group to remain
+    	log1.getProperties().clear();
+    	log1 = ITUtilLogs.assertUpdateLog(log1);
+    	log2 = ITUtilLogs.assertRetrieveLog("/" + log2.getId());
+    	assertEquals(1, log1.getProperties().size());
```

Comment:
It's good we check the logbook group property remains, however it may be more effective to also verify that other properties are operated on as expected? E.g. if the other properties are left alone but we attempt to change or delete the log group?

Reply:
I guess the other properties may be part of the update which makes it not known which changes to keep and which to stop.